more_autograding_examples/jupyter_notebook_autograding/config/test_input/grader.py

```diff
@@ -0,0 +1,78 @@
+from pathlib import Path
+import nbformat
+from nbconvert.preprocessors import ClearOutputPreprocessor, ExecutePreprocessor, CellExecutionError
+import base64
+
+# Execution API: https://nbconvert.readthedocs.io/en/latest/execute_api.html
+def execute_notebook(timeout=600):
+    
+    notebook_files = list(Path('.').glob('*.ipynb'))
+    if not notebook_files:
+        raise FileNotFoundError("No Jupyter notebook files found in the current directory.")
+    notebook_filename = Path(notebook_files[0])
+
+    with open(notebook_filename) as f:
+        # Load the notebook and do not convert it to a specific version
+        nb = nbformat.read(f, as_version=nbformat.NO_CONVERT)
+
+    # Clear output of notebook
+    cop = ClearOutputPreprocessor()
+    nb, _ = cop.preprocess(nb, {})
+
+    # Sets preprocessor with the following parameters:
+    # - timeout: maximum time in seconds to execute a cell
+    # - kernel_name: the kernel to use for execution (e.g., 'python3')
+    # - allow_errors: if True, allows the execution to continue even if a cell raises an error (Optional, defaults to False)
+    ep = ExecutePreprocessor(timeout=timeout, kernel_name='python3', allow_errors=False)
```

Comment:
I think we should probably use 'allow errors = true' so we can do partial credit for incomplete work (e.g., student skips 'part 2' of the assignment, but does complete 'part 3' that is independent of part 2.)

but we should have a visible error message surfaced to the submitty UI if there was an error on one or more steps of execution -- in addition to rendering the executed notebook.